lib/models/BaseSequelize.ts

```diff
@@ -19,21 +20,25 @@ import {BaseAssociation} from './association/BaseAssociation';
  * "classes" cannot extend ES6 classes
  */
 export abstract class BaseSequelize {
-
   throughMap: { [through: string]: any } = {};
-  _: { [modelName: string]: (typeof Model) } = {};
+  _: { [modelName: string]: typeof Model } = {};
+  _repos: {};
+  repositoryMode: boolean;
 
   static isISequelizeDbNameConfig(obj: any): obj is ISequelizeDbNameConfig {
-    return obj.hasOwnProperty("name") && obj.hasOwnProperty("username");
+    return obj.hasOwnProperty('name') && obj.hasOwnProperty('username');
   }
 
   static isISequelizeUriConfig(obj: any): obj is ISequelizeUriConfig {
-    return obj.hasOwnProperty("url");
+    return obj.hasOwnProperty('url');
   }
 
   static extend(target: any): void {
-
+    const _addModels = target.prototype.addModels;
     extend(target, this);
+    if (_addModels) {
+      target.prototype.addModels = _addModels;
+    }
   }
```

Comment:
Can you explain the purpose of this? Is there a case where it can be happen that `addModels` doesn't exist anymore after `extend` is called?

Reply:
i implement the addModels function in v4 to take into account the repositoryMode, but the extend override the v4 function with the one on BaseSequelize, so if the v4/v3 have that function implemented i keep that one and not the one on BaseSequelize.

the reason to implement the addModels in v4 and not change the one on BaseSequelize, was because the project in general use es5 and then this code

```
export function ModelFactory(model: any): any {
  return class extends model {};
}
```

compile into something that doesn't work, ( no classes in es5 ), so i decided to implement that in Sequelize v4 since that file is compiled to es6

When you drop support of v3, that patch could be removed.

If you have a better way to create the derived class of the model that work in es5 , let me know, so i can remove the addModels from v4 and modified the one on BaseSequelize.

Reply:
definitely a better idea